test/framework/alltypes_helpers.go

```diff
@@ -132,38 +131,47 @@ func DumpAllResources(ctx context.Context, input DumpAllResourcesInput) {
 	}
 }
 
-// DumpKubeSystemPodsForClusterInput is the input for DumpKubeSystemPodsForCluster.
-type DumpKubeSystemPodsForClusterInput struct {
-	Lister  Lister
-	LogPath string
-	Cluster *clusterv1.Cluster
+// DumpVersionAndKind specifies a version and kind which will be dumped
+type DumpVersionAndKind struct {
+	APIVersion string
+	Kind       string
+	Namespace  string
 }
```

Comment:
```suggestion
type DumpNamespaceAndGVK struct {
	GVK schema.GroupVersionKind
	Namespace  string
}
```

Let's use GVK and also set the Group in the Dump function.
This way we can use it for any kind of resource, not just the core api.